### PR TITLE
FIX: vat management for credit notes (and negative quantities)

### PR DESCRIPTION
**Pull request by altairis-tof**
in vat reports, vat was positive even if total_ht was negative, so vat reports were false (case of credit notes)

### CODE REVIEW

**Comment by altairis-tof** [comment]
TODO: find why total_vat is stored in absolute value in invoice table

**Comment by altairis-tof** [comment]
i got it ! when you edit a credit note line, if you put a positive price, this price is converted to negative but vat remains positive !

File: htdocs/compta/facture/class/facture.class.php [hunk 1]
```diff
@@ -2386,9 +2386,9 @@ function updateline($rowid, $desc, $pu, $qty, $remise_percent, $date_start, $dat
 			$this->line->date_start			= $date_start;
 			$this->line->date_end			= $date_end;
 			$this->line->total_ht			= (($this->type==self::TYPE_CREDIT_NOTE||$qty<0)?-abs($total_ht):$total_ht);  // For credit note and if qty is negative, total is negative
-			$this->line->total_tva			= $total_tva;
-			$this->line->total_localtax1	= $total_localtax1;
-			$this->line->total_localtax2	= $total_localtax2;
+			$this->line->total_tva			= (($this->type==self::TYPE_CREDIT_NOTE||$qty<0)?-abs($total_tva):$total_tva);  // For credit note and if qty is negative, total is negative
```
Comment (eldy) [inline htdocs/compta/facture/class/facture.class.php:2389]: Fix into function addline looks good to me. However this part seems to be here to manage corrupted data in base due to old bug ? Right ? If i am right, it should be fixed by a sql request and not into core code. For example by adding the sql that fix data into file repair.sql

**Comment by altairis-tof** [comment]
i disagree.

**Pull request closed without merging**